tuist/tuist
# Fix for incorrect bundle when using generated resource accessors

resolves: https://github.com/tuist/tuist/issues/4257

### Short description 📝

The bundle used within the generated resource accessor stencil template was incorrect for static frameworks.

### Details

- When accessing resources from static products, their dedicated resources bundle needs to be referenced 
- Tuist generates a bundle accessor `Bundle+<TargetName>.swift` to address is
- The templates used to generate the resource accessors (e.g. fonts, assets, etc...) need to reference this bundle rather than the default bundle to cater for the scenarios where static frameworks + resource bundles are used
- The templates access a `bundle` parameter, however it was missing! causing the fallback to the default `Bundle` in the templates which only works with dynamic frameworks or applications
- The bundle name is now passed along to the templates

Note ⚠️: There is a small caveat, in the case where bundle accessors are disabled (the code that accesses finds / accesses the bundle) - in that case we omit the bundle name as it won't be present and would have otherwise caused a compilation failure.

### How to test the changes locally 🧐

- Generate the static frameworks fixture

```sh
swift build
swift run tuist generate --path ./projects/tuist/fixtures/ios_app_with_static_frameworks_with_resources
```

- Verify the generate resource accessors for Module `A` reference the correct bundle (`Bundle.module`)
- Build and run the application
- Verify the custom font file loads on screen without any issues

### Checklist ✅

-  ~In case the PR introduces changes that affect users, the documentation has been updated.~
- ~In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.~
